File: c3/kart.cpp
#include <iostream>
#include <iomanip>
#include <conio.h>
#include <cmath>
#include <windows.h>
#include <ctime>
#include <string>

using namespace std;

void gotoxy(int x, int y)
{
    COORD cord;
    cord.X = x;
    cord.Y = y;
    SetConsoleCursorPosition(GetStdHandle(STD_OUTPUT_HANDLE), cord);
}

int input(string g){
	bool k = false;
	int x = 0;
	do{
		cout<<g;cin>>x;
		if(cin.fail()){cout<<"wartosc nieprawidlowa\n";}
		else{k = true;}
		cin.clear();cin.sync();
	}while(k==false);
	return x;
}

string sprawdz(int a , int b){
	if(a<b){
		return "ZA MALO";
	}
	else if(a>b){
		return "ZA DUZO";
	}
	else{
		return "TO JEST TO";
	}
}

void napisz(int x, int y, string z){
	gotoxy(x,y);
	cout<<z;
}

int los(int x){
	int r=1+rand()%(x-2);
	return r;
}

int main(){
    bool re = true;
	do{
		srand(time(NULL));
        system("cls");
        int l = los(100);
        bool d =false;
        while(d==false){
			int u = input("\npodaj liczbe calkowita:");
	        system("cls");
	        napisz(0,0,sprawdz(u,l));
	        if(u==l){d=true;}
    	}
		cout<<"\njeszcze raz? (T/N)"<<endl;
        if (tolower(getch())!='t'){re = false;}
    }while(re==true);
    return 0;
}

File: p3/kart.py
import random as r
import os as o
import msvcrt as ms

def gotoxy(x,y):
    print(f"\x1b[{y+1};{x+1}H",end='')

def los(a):
    return r.randint(1,a)

def inputh(g):
    k = False
    x = 0
    while True:
        try:
            x = int(input(g))
            k = True
        except ValueError:
            print("wariosc nieprawidlowa")
        if k:
            break 
    return x

def sprawdz(a,b):
    if a<b:
        return "ZA MALO"
    elif a>b:
        return 'ZA DUZO'
    else:
        return 'TO JEST TO'

def napisz(x,y,z):
    gotoxy(x,y)
    print(z)

def main():
    while True:
        o.system("cls")
        l = los(100)
        u = -1
        while u != l:
            u = inputh("podaj liczbe calkowita:")
            o.system("cls")
            napisz(0,0,sprawdz(u,l))
        print("czy chcesz powtorzyc program (T/N)")
        k = ms.getch().lower()
        if k != b't':
            break

if __name__=="__main__": 
    main()
